presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestReorderJoins.java

```diff
@@ -52,12 +61,139 @@ public void testChoosesAJoin()
                 .matches(join(
                         JoinNode.Type.INNER,
                         ImmutableList.of(equiJoinClause("B1", "C1"), equiJoinClause("A1", "C1")),
-                        join(
+                        Optional.empty(),
+                        Optional.of(PARTITIONED), join(
                                 JoinNode.Type.INNER,
                                 ImmutableList.of(equiJoinClause("A1", "B1")),
-                                values(ImmutableMap.of("A1", 0)),
-                                values(ImmutableMap.of("B1", 0)
-                                )),
-                        values(ImmutableMap.of("C1", 0))));
+                                Optional.empty(),
+                                Optional.of(PARTITIONED), values(ImmutableMap.of("A1", 0)),
+                                values(ImmutableMap.of("B1", 0))
+                        ),
+                        values(ImmutableMap.of("C1", 0))
+                ));
+    }
+
+    @Test
+    public void testReplicatesAndFlipsWhenOneTableMuchSmaller()
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("local")
+                .setSchema("tiny")
+                .setSystemProperty("join_distribution_type", "automatic")
+                .build();
+        LocalQueryRunner queryRunner = queryRunnerWithDummyNodeCount(session, 4);
+        new RuleTester(queryRunner)
+                .assertThat(new ReorderJoins(new CostComparator(1, 1, 1)))
+                .on(p ->
+                        p.joinGraph(
+                                ImmutableList.of(
+                                        p.values(new PlanNodeId("values1"), p.symbol("A1", BIGINT)),
+                                        p.values(new PlanNodeId("values2"), p.symbol("B1", BIGINT))),
+                                ImmutableList.of(
+                                        new JoinNode.EquiJoinClause(p.symbol("A1", BIGINT), p.symbol("B1", BIGINT))),
+                                ImmutableList.of()))
+                .withStats(ImmutableMap.of(
+                        new PlanNodeId("values1"), PlanNodeStatsEstimate.builder().setOutputRowCount(new Estimate(100)).build(),
+                        new PlanNodeId("values2"), PlanNodeStatsEstimate.builder().setOutputRowCount(new Estimate(10000)).build()))
+                .matches(join(
+                        JoinNode.Type.INNER,
+                        ImmutableList.of(equiJoinClause("B1", "A1")),
+                        Optional.empty(),
+                        Optional.of(REPLICATED), values(ImmutableMap.of("B1", 0)),
```

Comment:
The `values()` call should be on a separate line.

EDIT: here and in all the other tests in this file.